models/weather_api_handler.py

```diff
@@ -64,8 +64,14 @@ def verify_api_response(self, apiResponse):
         """tests if the response is valid"""
         if apiResponse is None:
             raise ApiCallNonResposive
-        if apiResponse.get("forecast").get("forecastday") is None:
-            apiResponse = {"error": "Could not retrieve weather data."}
+        if apiResponse.get("forecast") is None or apiResponse.get("forecast").get("forecastday") is None:
```

Comment:
Maybe these lines of code could be:
```
try:
    apiResponseCheck = apiResponse.get("forecast").get("forecastday")[-1].get("day")
except (AttributeError, IndexError):
    raise ApiReturnNoneResults
if apiResponseCheck is None or apiResponseCheck == {}:
    raise ApiReturnNoneResults
```
It looks a little cleaner, rather than 5 if statements where 4 of them are just checking if parts of the return statement exist.